devito/types/grid.py

```diff
@@ -546,6 +633,16 @@ def __subdomain_finalize__(self, grid, **kwargs):
         self._dimensions = tuple(sub_dimensions)
         self._dtype = grid.dtype
 
+    @property
+    def shape_local(self):
+        return self._shape_local
```

Comment:
If no validation or construction is being done here, why can't `shape_local` just be a public attribute?

Reply:
To prevent users from accidentally manipulating the underlying values

Reply:
~I guess, but that's only the case if `self._shape_local` is used internally~
I forgot that if you don't have a setter it won't overwrite the property